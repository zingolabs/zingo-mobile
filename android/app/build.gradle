apply plugin: "com.android.application"
apply plugin: "com.facebook.react"
apply plugin: 'kotlin-android'

/**
 * This is the configuration block to customize your React Native Android app.
 * By default you don't need to apply any configuration, just uncomment the lines you need.
 */

react {
    /* Folders */
    //   The root of your project, i.e. where "package.json" lives. Default is '..'
    // root = file("../")
    //   The folder where the react-native NPM package is. Default is ../node_modules/react-native
    // reactNativeDir = file("../node_modules/react-native")
    //   The folder where the react-native Codegen package is. Default is ../node_modules/@react-native/codegen
    // codegenDir = file("../node_modules/@react-native/codegen")
    //   The cli.js file which is the React Native CLI entrypoint. Default is ../node_modules/react-native/cli.js
    // cliFile = file("../node_modules/react-native/cli.js")
    /* Variants */
    //   The list of variants to that are debuggable. For those we're going to
    //   skip the bundling of the JS bundle and the assets. By default is just 'debug'.
    //   If you add flavors like lite, prod, etc. you'll have to list your debuggableVariants.
    // debuggableVariants = ["liteDebug", "prodDebug"]
    /* Bundling */
    //   A list containing the node command and its flags. Default is just 'node'.
    // nodeExecutableAndArgs = ["node"]
    //
    //   The command to run when bundling. By default is 'bundle'
    // bundleCommand = "ram-bundle"
    //
    //   The path to the CLI configuration file. Default is empty.
    // bundleConfig = file(../rn-cli.config.js)
    //
    //   The name of the generated asset file containing your JS bundle
    // bundleAssetName = "MyApplication.android.bundle"
    //
    //   The entry file for bundle generation. Default is 'index.android.js' or 'index.js'
    // entryFile = file("../js/MyApplication.android.js")
    //
    //   A list of extra flags to pass to the 'bundle' commands.
    //   See https://github.com/react-native-community/cli/blob/main/docs/commands.md#bundle
    // extraPackagerArgs = []
    /* Hermes Commands */
    //   The hermes compiler command to run. By default it is 'hermesc'
    // hermesCommand = "$rootDir/my-custom-hermesc/bin/hermesc"
    //
    //   The list of flags to pass to the Hermes compiler. By default is "-O", "-output-source-map"
    // hermesFlags = ["-O", "-output-source-map"]
}

/**
 * Set this to true to Run Proguard on Release builds to minify the Java bytecode.
 */
def enableProguardInReleaseBuilds = false

/**
 * The preferred build flavor of JavaScriptCore (JSC)
 *
 * For example, to use the international variant, you can use:
 * `def jscFlavor = 'org.webkit:android-jsc-intl:+'`
 *
 * The international variant includes ICU i18n library and necessary data
 * allowing to use e.g. `Date.toLocaleString` and `String.localeCompare` that
 * give correct results when using with locales other than en-US.  Note that
 * this variant is about 6MiB larger per architecture than default.
 */
def jscFlavor = 'org.webkit:android-jsc:+'

/**
 * Whether to enable building a separate APK for each ABI.
 *
 * Defaults to false but can be set to true with the project properties flag
 * e.g. ./gradlew assembleDebug -PsplitApk=true
 *
 * Additional option to include a universal APK
 * e.g. ./gradlew assembleRelease -PsplitApk=true -PincludeUniversalApk=true
 */
ext.splitApk = project.properties['splitApk'] ?: false
ext.includeUniversalApk = project.properties['includeUniversalApk'] ?: false

android {
    namespace 'org.ZingoLabs.Zingo'
    ndkVersion rootProject.ext.ndkVersion

    compileSdkVersion rootProject.ext.compileSdkVersion

    compileOptions {
        sourceCompatibility = "18.0.2.1"
        targetCompatibility = "18.0.2.1"
    }

    kotlinOptions {
        jvmTarget = '18'
    }

    defaultConfig {
        applicationId 'org.ZingoLabs.Zingo' // Real
        //applicationId 'com.ZingoMobile' // @Test
        minSdkVersion rootProject.ext.minSdkVersion
        targetSdkVersion rootProject.ext.targetSdkVersion
        versionCode 126 // Real
        //versionCode 117 // @Test
        versionName "zingo-1.2.5" // Real
        missingDimensionStrategy 'react-native-camera', 'general'
        testBuildType System.getProperty('testBuildType', 'debug')
        testInstrumentationRunner 'androidx.test.runner.AndroidJUnitRunner'
    }

    splits {
        abi {
            enable splitApk.toBoolean()
            reset()
            include "armeabi-v7a", "x86", "arm64-v8a", "x86_64"
            universalApk includeUniversalApk.toBoolean()
        }
    }

    signingConfigs {
        debug {
            storeFile file('debug.keystore')
            storePassword 'android'
            keyAlias 'androiddebugkey'
            keyPassword 'android'
        }
    }

    packagingOptions {
        jniLibs {
            pickFirsts += ['lib/armeabi-v7a/libc++_shared.so', 'lib/arm64-v8a/libc++_shared.so', 'lib/x86/libc++_shared.so', 'lib/x86_64/libc++_shared.so']
        }
    }

    buildTypes {
        debug {
            signingConfig signingConfigs.debug
        }
        release {
            // Caution! In production, you need to generate your own keystore file.
            // see https://reactnative.dev/docs/signed-apk-android.
            signingConfig signingConfigs.debug
            minifyEnabled enableProguardInReleaseBuilds
            proguardFiles getDefaultProguardFile("proguard-android.txt"), "proguard-rules.pro"
            proguardFile "${rootProject.projectDir}/../node_modules/detox/android/detox/proguard-rules-app.pro"
        }
    }

    buildFeatures {
        viewBinding = true
    }

    testOptions {
        // Managed devices for gradle integration testing
        managedDevices {
            devices {
                pixel2api29_x86 (com.android.build.api.dsl.ManagedVirtualDevice) {
                    device = "Pixel 2"
                    apiLevel = 29
                    systemImageSource = "aosp"
                    require64Bit = false
                }
                pixel2api30_x86_64 (com.android.build.api.dsl.ManagedVirtualDevice) {
                    device = "Pixel 2"
                    apiLevel = 30
                    systemImageSource = "aosp"
                    require64Bit = true
                }
            }
            groups {
                x86_Archs {
                    targetDevices.add(devices.pixel2api29_x86)
                    targetDevices.add(devices.pixel2api30_x86_64)
                }
            }
        }
    }
    
    // // applicationVariants are e.g. debug, release
    // applicationVariants.all { variant ->
    //     variant.outputs.each { output ->
    //         // For each separate APK per architecture, set a unique version code as described here:
    //         // https://developer.android.com/studio/build/configure-apk-splits.html
    //         // Example: versionCode 1 will generate 1001 for armeabi-v7a, 1002 for x86, etc.
    //         def versionCodes = ["armeabi-v7a": 1, "x86": 2, "arm64-v8a": 3, "x86_64": 4]
    //         def abi = output.getFilter(OutputFile.ABI)
    //         if (abi != null) {  // null for the universal-debug, universal-release variants
    //             output.versionCodeOverride =
    //                     defaultConfig.versionCode * 1000 + versionCodes.get(abi)
    //         }
    //     }
    // }
}


// Map for the version code that gives each ABI a value.
ext.abiCodes = ['armeabi-v7a':1, 'x86':2, 'arm64-v8a': 3, 'x86_64':4]

import com.android.build.OutputFile

// For each APK output variant, override versionCode with a combination of
// ext.abiCodes * 10000 + variant.versionCode. 
// variant.versionCode is equal to defaultConfig.versionCode.
// If you configure product flavors that define their own versionCode, 
// variant.versionCode uses that value instead.
android.applicationVariants.all { variant ->
  variant.outputs.each { output ->
    def baseAbiVersionCode = project.ext.abiCodes.get(output.getFilter(OutputFile.ABI))

    // Because abiCodes.get() returns null for ABIs that are not mapped by ext.abiCodes,
    // the following code doesn't override the version code for universal APKs.
    // However, because you want universal APKs to have the lowest version code,
    // this outcome is desirable.
    if (baseAbiVersionCode != null) {

      // Assigns the new version code to versionCodeOverride, which changes the
      // version code for only the output APK, not for the variant itself. Skipping
      // this step causes Gradle to use the value of variant.versionCode for the APK.
      output.versionCodeOverride = baseAbiVersionCode * 10000 + variant.versionCode
    }
  }
}
            
dependencies {
    // The version of react-native is set by the React Native Gradle Plugin
    implementation("com.facebook.react:react-android")
    androidTestImplementation('com.wix:detox:+')
    implementation 'androidx.appcompat:appcompat:1.1.0'
    implementation fileTree(dir: "libs", include: ["*.jar"])

    implementation "androidx.swiperefreshlayout:swiperefreshlayout:1.0.0"
    implementation 'com.facebook.soloader:soloader:0.10.5+'

    debugImplementation("com.facebook.flipper:flipper:${FLIPPER_VERSION}") {
      exclude group:'com.facebook.fbjni'
    }

    // Detox tests getAttributes() needs this
    debugImplementation('com.google.android.material:material:1.2.1')

    debugImplementation("com.facebook.flipper:flipper-network-plugin:${FLIPPER_VERSION}") {
        exclude group:'com.facebook.flipper'
        exclude group:'com.squareup.okhttp3', module:'okhttp'
    }

    debugImplementation("com.facebook.flipper:flipper-fresco-plugin:${FLIPPER_VERSION}") {
        exclude group:'com.facebook.flipper'
    }

    if (hermesEnabled) {
        implementation("com.facebook.react:hermes-engine:+") {
            exclude group:'com.facebook.fbjni'
        }
    } else {
        implementation jscFlavor
    }
	implementation "org.jetbrains.kotlin:kotlin-stdlib:$kotlinVersion"

    def work_version = "2.7.1"

    // (Java only)
    implementation "androidx.work:work-runtime:$work_version"

    // Kotlin + coroutines
    implementation "androidx.work:work-runtime-ktx:$work_version"

    // optional - RxJava2 support
    implementation "androidx.work:work-rxjava2:$work_version"

    // optional - GCMNetworkManager support
    implementation "androidx.work:work-gcm:$work_version"

    // optional - Test helpers
    androidTestImplementation "androidx.work:work-testing:$work_version"

    // optional - Multiprocess support
    implementation "androidx.work:work-multiprocess:$work_version"
    
    // google truth testing framework
    androidTestImplementation "com.google.truth:truth:1.1.3"
    
    // JSON parsing
    implementation "com.fasterxml.jackson.module:jackson-module-kotlin:2.14.+"
    
    // JUnit test runners
    androidTestImplementation 'junit:junit:4.13'
}

// Run this once to be able to run the application with BUCK
// puts all compile dependencies into folder libs for BUCK to use
task copyDownloadableDepsToLibs(type: Copy) {
    from configurations.implementation
    into 'libs'
}

apply from: file("../../node_modules/@react-native-community/cli-platform-android/native_modules.gradle"); applyNativeModulesAppBuildGradle(project)
repositories {
    mavenCentral()
    google()
}
