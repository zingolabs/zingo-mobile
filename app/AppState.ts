// new class with all the information about the sync process
export class SyncStatusReport {
  // sync ID
  syncID: number;

  // total of batches to process (usually 1000 blocks per batch)
  totalBatches: number;

  // batch that is processing now
  currentBatch: number;

  // total blocks per batch
  blocksPerBatch: number;

  // last block of the wallet
  lastBlockWallet: number;

  // block that is processing
  currentBlock: number;

  inProgress: boolean;

  lastError: string;

  secondsPerBatch: number;

  percent: number;

  message: string;

  process_end_block: number;

  lastBlockServer: number;

  constructor() {
    this.syncID = 0;
    this.totalBatches = 0;
    this.currentBatch = 0;
    this.blocksPerBatch = 1000; // this is set in zingolib
    this.lastBlockWallet = 0;
    this.currentBlock = 0;
    this.inProgress = false;
    this.lastError = '';
    this.secondsPerBatch = 0;
    this.percent = 0;
    this.message = '';
    this.process_end_block = 0;
    this.lastBlockServer = 0;
  }
}

export class TotalBalance {
  // Total t address, confirmed and spendable
  transparentBal: number;

  // Total private, confirmed + unconfirmed
  privateBal: number;

  // Total orchard, confirmed + unconfirmed
  orchardBal: number;

  // Total private, confirmed funds that are spendable
  spendablePrivate: number;

  // Total orchard, confirmed funds that are spendable
  spendableOrchard: number;

  // Total unconfirmed + spendable
  total: number;

  constructor() {
    this.transparentBal = 0;
    this.privateBal = 0;
    this.spendablePrivate = 0;
    this.orchardBal = 0;
    this.spendableOrchard = 0;
    this.total = 0;
  }
}

export class Address {
  uaAddress: string;
  address: string;
  addressKind: string;
  containsPending: boolean;

  constructor(uaAddress: string, address: string, addressKind: string) {
    this.uaAddress = uaAddress;
    this.address = address;
    this.addressKind = addressKind;
    this.containsPending = false;
  }
}

export class AddressBookEntry {
  label: string;

  address: string;

  constructor(label: string, address: string) {
    this.label = label;
    this.address = address;
  }
}

export class SettingsFileEntry {
  server?: string;

  constructor(server: string) {
    this.server = server;
  }
}

export interface TxDetailType {
  address: string;
  amount: number;
  memo: string | null;
}

// List of transactions.
export interface Transaction {
  type: string;
  address: string;
  amount: number;
  position: string;
  confirmations: number;
  txid: string;
  time: number;
  zec_price: number | null;
  detailedTxns: TxDetailType[];
}

// This is the type that the RPC interface expects for sending
export interface SendJsonToType {
  address: string;
  amount: number;
  memo?: string;
}

export class ToAddr {
  id: number;
  to: string;
  amount: string;
  amountUSD: string;
  memo: string;

  constructor(id: number) {
    this.id = id;

    this.to = '';
    this.amount = '';
    this.amountUSD = '';
    this.memo = '';
  }
}

export class SendPageState {
  toaddrs: ToAddr[];

  constructor() {
    this.toaddrs = [];
  }
}

export class ReceivePageState {
  // A newly created address to show by default
  newAddress: string;

  // The key used for the receive page component.
  // Increment to force re-render
  rerenderKey: number;

  constructor() {
    this.newAddress = '';
    this.rerenderKey = 0;
  }
}

export interface InfoType {
  chain_name: string;
  serverUri: string;
  latestBlock: number;
  connections: number;
  version: string;
  verificationProgress: number;
  currencyName: string;
  solps: number;
  zecPrice: number | null;
  defaultFee: number;
}

export class SendProgress {
  sendInProgress: boolean;
  progress: number;
  total: number;
  etaSeconds: number | string;

  constructor() {
    this.sendInProgress = false;
    this.progress = 0;
    this.total = 0;
    this.etaSeconds = 0;
  }
}

export interface ServerSelectState {
  modalIsOpen: boolean;
}

export class PasswordState {
  showPassword: boolean;

  confirmNeeded: boolean;

  passwordCallback: ((password: string) => void) | null;

  closeCallback: (() => void) | null;

  helpText: string | null;

  constructor() {
    this.showPassword = false;
    this.confirmNeeded = false;
    this.passwordCallback = null;
    this.closeCallback = null;
    this.helpText = null;
  }
}

export class ErrorModalData {
  title: string;
  body: string;

  modalIsOpen: boolean;

  constructor() {
    this.modalIsOpen = false;
    this.title = '';
    this.body = '';
  }
}

export interface ConnectedCompanionApp {
  name: string;
  lastSeen: number;
}

export interface WalletSeed {
  seed: string;
  birthday: number;
}

export interface SyncStatus {
  inProgress: boolean;
  progress: number;
  blocks: string;
}

export class WalletSettings {
  download_memos: string;
  transaction_filter_threshold: string;
  server?: string;

  constructor() {
    this.download_memos = 'wallet';
    this.transaction_filter_threshold = '500';
  }
}

export default interface AppState {
  // Info about the current sync process
  syncStatusReport: SyncStatusReport;

  // The total confirmed and unconfirmed balance in this wallet
  totalBalance: TotalBalance;

  // A map type that contains address -> privatekey mapping, for display on the receive page
  // This mapping is ephemeral, and will disappear when the user navigates away.
  addressPrivateKeys: Map<string, string>;

  // List of all addresses in the wallet, including change addresses and addresses
  // that don't have any balance or are unused
  addresses: Address[];

  // List of Address / Label pairs
  addressBook: AddressBookEntry[];

  // List of all T and Z and O transactions
  transactions: Transaction[] | null;

  // The state of the send page, as the user constructs a transaction
  sendPageState: SendPageState;

  // Any state for the receive page
  receivePageState: ReceivePageState;

  // getinfo and getblockchaininfo result
  info: InfoType | null;

  // Is the app rescanning?
  rescanning: boolean;

  // Callbacks for the password dialog box
  //passwordState: PasswordState;

  wallet_settings: WalletSettings;

  syncingStatus: SyncStatus | null;

  // Data for any error or info modal
  errorModalData: ErrorModalData;

  // Build progress from Tx
  txBuildProgress: SendProgress;

  walletSeed: WalletSeed | null;

  isMenuDrawerOpen: boolean;

  selectedMenuDrawerItem: string;

  aboutModalVisible: boolean;

  computingModalVisible: boolean;

  settingsModalVisible: boolean;

  infoModalVisible: boolean;

  rescanModalVisible: boolean;

  seedViewModalVisible: boolean;
  seedChangeModalVisible: boolean;
  seedBackupModalVisible: boolean;
  seedServerModalVisible: boolean;

  syncReportModalVisible: boolean;

  newServer: string | null;

  uaAddress: string | null;

  // eslint-disable-next-line semi
}
