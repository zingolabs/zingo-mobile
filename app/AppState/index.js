import AddressClass from './classes/AddressClass';
import AddressBookClass from './classes/AddressBookClass';
import ErrorModalDataClass from './classes/ErrorModalDataClass';
import ReceivePageStateClass from './classes/ReceivePageStateClass';
import SendPageStateClass from './classes/SendPageStateClass';
import SendProgressClass from './classes/SendProgressClass';
import SettingsFileClass from './classes/SettingsFileClass';
import SyncingStatusClass from './classes/SyncingStatusClass';
import ToAddrClass from './classes/ToAddrClass';
import TotalBalanceClass from './classes/TotalBalanceClass';
import WalletSettingsClass from './classes/WalletSettingsClass';

import InfoType from './types/InfoType';
import SendJsonToTypeType from './types/SendJsonToTypeType';
import TransactionType from './types/TransactionType';
import TxDetailType from './types/TxDetailType';
import WalletType from './types/WalletType';
import { TranslateType } from './types/TranslateType';
import NetInfoType from './types/NetInfoType';
import BackgroundType from './types/BackgroundType';
import BackgroundErrorType from './types/BackgroundErrorType';
import ServerType from './types/ServerType';
import SnackbarType from './types/SnackbarType';
import ZecPriceType from './types/ZecPriceType';
import FetchErrorType from './types/FetchErrorType';

import AppStateLoaded from './AppStateLoaded';
import AppStateLoading from './AppStateLoading';

export {
  AppStateLoaded,
  AppStateLoading,
  AddressClass,
  AddressBookClass,
  ErrorModalDataClass,
  ReceivePageStateClass,
  SendPageStateClass,
  SendProgressClass,
  SettingsFileClass,
  SyncingStatusClass,
  ToAddrClass,
  TotalBalanceClass,
  WalletSettingsClass,
  TransactionType,
  TxDetailType,
  WalletType,
  SendJsonToTypeType,
  InfoType,
  ZecPriceType,
  BackgroundType,
  BackgroundErrorType,
  TranslateType,
  NetInfoType,
  ServerType,
  SnackbarType,
  FetchErrorType,
};
