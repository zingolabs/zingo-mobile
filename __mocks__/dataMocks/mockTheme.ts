import { ThemeType } from '../../app/types';

export const mockTheme: ThemeType = {
  dark: true,
  colors: {
    background: '#011401', //'#010101',
    card: '#011401', //'#401717',
    border: '#ffffff',
    primary: '#18bd18', //'#df4100',
    primaryDisabled: '#5a8c5a', //'rgba(90, 140, 90, 1)',
    secondaryDisabled: '#233623',
    text: '#c3c3c3',
    zingo: '#888888',
    placeholder: '#888888',
    money: '#ffffff',
    syncing: '#ebff5a',
    notification: '',
    sideMenuBackground: '#040C17',
  },
};
