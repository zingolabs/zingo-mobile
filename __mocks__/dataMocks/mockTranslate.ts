import { CurrencyEnum, DownloadMemosEnum, LanguageEnum } from '../../app/AppState';

export const mockTranslate = (p: string) => {
  if (p === 'about.copyright') {
    return String([
      '1 text translated line 1',
      '2 text translated line 2',
      '3 text translated line 3',
      '4 text translated line 4',
      '5 text translated line 5',
    ]);
  } else if (p === 'seed.buttontexts') {
    return `{
        "new": ["new"],
        "change": ["change"],
        "server": ["server"],
        "view": ["view"],
        "restore": ["restore"],
        "backup": ["backup"]
      }`;
  } else if (p === 'settings.memos') {
    return [
      {
        value: DownloadMemosEnum.noneMemos,
        text: 'text none',
      },
      {
        value: DownloadMemosEnum.walletMemos,
        text: 'text wallet',
      },
      {
        value: DownloadMemosEnum.allMemos,
        text: 'text all',
      },
    ];
  }
  if (p === 'settings.currencies') {
    return [
      {
        value: '',
        text: 'text no currency',
      },
      {
        value: CurrencyEnum.USDCurrency,
        text: 'text USD',
      },
    ];
  }
  if (p === 'settings.languages') {
    return [
      {
        value: LanguageEnum.en,
        text: 'text en',
      },
      {
        value: LanguageEnum.es,
        text: 'text es',
      },
      {
        value: LanguageEnum.pt,
        text: 'text pt',
      },
      {
        value: LanguageEnum.ru,
        text: 'text ru',
      },
    ];
  }
  if (p === 'settings.sendalls') {
    return [
      {
        value: true,
        text: 'text true',
      },
      {
        value: false,
        text: 'text false',
      },
    ];
  }
  if (p === 'settings.donations') {
    return [
      {
        value: true,
        text: 'text true',
      },
      {
        value: false,
        text: 'text false',
      },
    ];
  }
  if (p === 'settings.rescanmenus') {
    return [
      {
        value: true,
        text: 'text true',
      },
      {
        value: false,
        text: 'text false',
      },
    ];
  } else if (p === 'settings.recoverywalletinfoondevices') {
    return [
      {
        value: true,
        text: 'text true',
      },
      {
        value: false,
        text: 'text false',
      },
    ];
  } else {
    return 'text translated';
  }
};
