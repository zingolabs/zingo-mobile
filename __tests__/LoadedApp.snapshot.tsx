/**
 * @format
 */

import 'react-native';
import React from 'react';

import { render } from '@testing-library/react-native';
import { LoadedAppClass } from '../app/LoadedApp';

// Importa el módulo I18n
// eslint-disable-next-line @typescript-eslint/no-unused-vars
import { I18n } from 'i18n-js';
import { LanguageEnum, ModeEnum, SelectServerEnum, CurrencyEnum } from '../app/AppState';
import { mockTheme } from '../__mocks__/dataMocks/mockTheme';
import { mockTranslate } from '../__mocks__/dataMocks/mockTranslate';
import { mockNavigation } from '../__mocks__/dataMocks/mockNavigation';
import { mockRoute } from '../__mocks__/dataMocks/mockRoute';
import { mockServer } from '../__mocks__/dataMocks/mockServer';
import { mockBackground } from '../__mocks__/dataMocks/mockBackground';
import { mockSecurity } from '../__mocks__/dataMocks/mockSecurity';
import { mockAddressBook } from '../__mocks__/dataMocks/mockAddressBook';

// Crea un mock para el constructor de I18n
jest.mock('i18n-js', () => ({
  __esModule: true,
  I18n: jest.fn().mockImplementation(() => ({
    t: jest.fn(),
    // Agrega otros métodos y propiedades según sea necesario para tus pruebas
  })),
}));

jest.useFakeTimers();
jest.mock('@fortawesome/react-native-fontawesome', () => ({
  FontAwesomeIcon: '',
}));
jest.mock('react-native-localize', () => ({
  getNumberFormatSettings: () => {
    return {
      decimalSeparator: '.',
      groupingSeparator: ',',
    };
  },
}));
jest.mock('react-native-tab-view', () => ({
  TabView: '',
  TabBar: '',
}));
jest.mock('react-native-option-menu', () => '');
jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper');
jest.mock('@react-native-community/netinfo', () => {
  return {
    addEventListener: jest.fn(),
    fetch: jest.fn().mockImplementation(() =>
      Promise.resolve({
        isConnected: true,
        isInternetReachable: true,
      }),
    ),
  };
});
jest.mock('react-native', () => {
  const RN = jest.requireActual('react-native');

  RN.NativeModules.RPCModule = {
    execute: jest.fn(() => '[]'),
    getValueTransfersList: jest.fn(() => '{ "value_transfers": [] }'),
  };

  return RN;
});
jest.mock('react-native-fs', () => ({
  readFile: jest.fn(() => Promise.resolve('{}')), // o Promise.reject(new Error('File not found'))
  writeFile: jest.fn(() => Promise.resolve()), // o Promise.reject(new Error('Write failed'))
  // Agrega más funciones mockeadas según sea necesario
}));
jest.mock('react-native-picker-select', () => 'RNPickerSelect');
jest.mock('react-native-device-info', () => ({
  getSystemName: jest.fn(() => 'Mocked System Name'),
  getSystemVersion: jest.fn(() => 'Mocked System Version'),
  getManufacturer: jest.fn(() => 'Mocked Manufacturer'),
  getModel: jest.fn(() => 'Mocked Model'),
}));

// test suite
describe('Component LoadedApp - test', () => {
  //snapshot test
  test('LoadedApp - snapshot', () => {
    const language = LanguageEnum.en;
    const currency = CurrencyEnum.noCurrency;
    const sendAll = false;
    const rescanMenu = false;
    const donation = false;
    const privacy = false;
    const mode = ModeEnum.basic;
    const readOnly = false;
    const toggleTheme = jest.fn();
    const selectServer = SelectServerEnum.auto;
    const loadedapp = render(
      <LoadedAppClass
        navigation={mockNavigation}
        route={mockRoute}
        toggleTheme={toggleTheme}
        translate={mockTranslate}
        theme={mockTheme}
        language={language}
        currency={currency}
        server={mockServer}
        sendAll={sendAll}
        donation={donation}
        privacy={privacy}
        mode={mode}
        background={mockBackground}
        readOnly={readOnly}
        addressBook={mockAddressBook}
        security={mockSecurity}
        selectServer={selectServer}
        rescanMenu={rescanMenu}
      />,
    );
    expect(loadedapp.toJSON()).toMatchSnapshot();
  });
});
