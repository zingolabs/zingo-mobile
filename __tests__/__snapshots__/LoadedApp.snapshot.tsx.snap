// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component LoadedApp - test LoadedApp - snapshot 1`] = `
<View
  onLayout={[Function]}
  style={
    {
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      [
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        {
          "right": 250,
        },
      ]
    }
  >
    <View
      style={
        {
          "height": "100%",
        }
      }
    >
      <RCTScrollView
        contentContainerStyle={
          {
            "display": "flex",
          }
        }
        scrollsToTop={false}
        style={
          {
            "backgroundColor": "#010101",
            "flex": 1,
            "height": 1334,
            "width": 750,
          }
        }
      >
        <View>
          <Text
            style={
              {
                "color": "rgb(28, 28, 30)",
                "fontSize": 18,
                "fontWeight": "600",
                "marginLeft": 30,
                "marginVertical": 10,
                "opacity": 1,
              }
            }
          >
            text translated
          </Text>
          <View
            style={
              {
                "backgroundColor": "rgb(0, 122, 255)",
                "height": 1,
              }
            }
          />
          <View
            style={
              {
                "display": "flex",
                "marginLeft": 20,
              }
            }
          >
            <Text
              onPress={[Function]}
              style={
                {
                  "color": "rgb(28, 28, 30)",
                  "fontSize": 14,
                  "fontWeight": "600",
                  "opacity": 1,
                  "paddingTop": 25,
                }
              }
              testID="menu.about"
            >
              text translated
            </Text>
            <Text
              onPress={[Function]}
              style={
                {
                  "color": "rgb(28, 28, 30)",
                  "fontSize": 14,
                  "fontWeight": "600",
                  "opacity": 1,
                  "paddingTop": 25,
                }
              }
              testID="menu.settings"
            >
              text translated
            </Text>
            <Text
              onPress={[Function]}
              style={
                {
                  "color": "rgb(28, 28, 30)",
                  "fontSize": 14,
                  "fontWeight": "600",
                  "opacity": 1,
                  "paddingTop": 25,
                }
              }
              testID="menu.addressbook"
            >
              text translated
            </Text>
            <Text
              onPress={[Function]}
              style={
                {
                  "color": "rgb(28, 28, 30)",
                  "fontSize": 14,
                  "fontWeight": "600",
                  "opacity": 1,
                  "paddingTop": 25,
                }
              }
              testID="menu.walletseedufvk"
            >
              text translated
            </Text>
            <Text
              onPress={[Function]}
              style={
                {
                  "color": "rgb(28, 28, 30)",
                  "fontSize": 14,
                  "fontWeight": "600",
                  "opacity": 1,
                  "paddingTop": 25,
                }
              }
              testID="menu.insight"
            >
              text translated
            </Text>
            <Text
              onPress={[Function]}
              style={
                {
                  "color": "rgb(28, 28, 30)",
                  "fontSize": 14,
                  "fontWeight": "600",
                  "opacity": 1,
                  "paddingTop": 25,
                }
              }
              testID="menu.tipzingolabs"
            >
              text translated
            </Text>
            <Text
              onPress={[Function]}
              style={
                {
                  "color": "rgb(28, 28, 30)",
                  "fontSize": 14,
                  "fontWeight": "600",
                  "opacity": 1,
                  "paddingTop": 25,
                }
              }
            >
              text translated
            </Text>
          </View>
        </View>
      </RCTScrollView>
      <View
        style={
          {
            "backgroundColor": "#010101",
            "bottom": 5,
            "flexDirection": "row",
            "padding": 10,
            "position": "absolute",
          }
        }
      >
        <Text
          style={
            {
              "color": "rgb(216, 216, 216)",
              "fontSize": 8,
            }
          }
        >
          Version : 
        </Text>
        <Text
          style={
            {
              "color": undefined,
              "fontSize": 8,
            }
          }
        >
          text translated
        </Text>
        <Text
          style={
            {
              "color": "rgb(216, 216, 216)",
              "fontSize": 8,
              "marginLeft": 10,
            }
          }
        >
          text translatedtext translated
        </Text>
      </View>
    </View>
  </View>
  <View
    collapsable={false}
    onMoveShouldSetResponder={[Function]}
    onMoveShouldSetResponderCapture={[Function]}
    onResponderEnd={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderReject={[Function]}
    onResponderRelease={[Function]}
    onResponderStart={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    style={
      {
        "backgroundColor": "transparent",
        "flex": 1,
        "height": 1334,
        "left": 0,
        "overflow": "hidden",
        "position": "absolute",
        "top": 0,
        "transform": [
          {
            "translateX": 0,
          },
        ],
        "width": 750,
      }
    }
  >
    <Modal
      animationType="slide"
      hardwareAccelerated={false}
      onRequestClose={[Function]}
      transparent={false}
      visible={false}
    />
    <Modal
      animationType="slide"
      hardwareAccelerated={false}
      onRequestClose={[Function]}
      transparent={false}
      visible={false}
    />
    <Modal
      animationType="slide"
      hardwareAccelerated={false}
      onRequestClose={[Function]}
      transparent={false}
      visible={false}
    />
    <Modal
      animationType="slide"
      hardwareAccelerated={false}
      onRequestClose={[Function]}
      transparent={false}
      visible={false}
    />
    <Modal
      animationType="slide"
      hardwareAccelerated={false}
      onRequestClose={[Function]}
      transparent={false}
      visible={false}
    />
    <Modal
      animationType="slide"
      hardwareAccelerated={false}
      onRequestClose={[Function]}
      transparent={false}
      visible={false}
    />
    <Modal
      animationType="slide"
      hardwareAccelerated={false}
      onRequestClose={[Function]}
      transparent={false}
      visible={false}
    />
    <Modal
      animationType="slide"
      hardwareAccelerated={false}
      onRequestClose={[Function]}
      transparent={false}
      visible={false}
    />
    <Modal
      animationType="slide"
      hardwareAccelerated={false}
      onRequestClose={[Function]}
      transparent={false}
      visible={false}
    />
    <Modal
      animationType="slide"
      hardwareAccelerated={false}
      onRequestClose={[Function]}
      transparent={false}
      visible={false}
    />
    <Modal
      animationType="slide"
      hardwareAccelerated={false}
      onRequestClose={[Function]}
      transparent={false}
      visible={false}
    />
    <Modal
      animationType="slide"
      hardwareAccelerated={false}
      onRequestClose={[Function]}
      transparent={false}
      visible={false}
    />
    <Modal
      animationType="slide"
      hardwareAccelerated={false}
      onRequestClose={[Function]}
      transparent={false}
      visible={false}
    />
    <Modal
      animationType="slide"
      hardwareAccelerated={false}
      onRequestClose={[Function]}
      transparent={false}
      visible={false}
    />
    <Modal
      animationType="slide"
      hardwareAccelerated={false}
      onRequestClose={[Function]}
      transparent={false}
      visible={false}
    />
    <Modal
      animationType="slide"
      hardwareAccelerated={false}
      onRequestClose={[Function]}
      transparent={false}
      visible={false}
    />
    <Modal
      animationType="slide"
      hardwareAccelerated={false}
      onRequestClose={[Function]}
      transparent={false}
      visible={false}
    />
    <View />
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#011401",
          "display": "flex",
          "height": "100%",
          "justifyContent": "center",
        }
      }
    >
      <ActivityIndicator
        color="#18bd18"
        size="large"
      />
    </View>
  </View>
</View>
`;
